### PR TITLE
fix(core): fix issues with UTF8 table and column names

### PR DESCRIPTION
**Pull request by bluestreak01**
- introduced API to read symbolic link targets
- `drop table`  of table on a volume (softlink) will remove both the link and the link target
- refactored `utf8decode` methods for clarity. Those convert utf8 to utf16 and often used inappropriately. I believe method name was contributing to the confusion
- replaced constant checks with fast path checks:

```
Benchmark                               Mode  Cnt   Score   Error  Units
CairoKeywordBenchmark.testConventional  avgt    5  62.926 ± 0.228  ns/op
CairoKeywordBenchmark.testFastPath      avgt    5   7.038 ± 0.089  ns/op
```

- refactored `ServerMainForeignTableTest` and its derivatives to remove file system access races.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 86 / 96 (89.58%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 162 / 182 (89.01%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalPurgeJob.java|3|7|42.86%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoKeywords.java|51|54|94.44%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 162 / 182 (89.01%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalPurgeJob.java|3|7|42.86%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoKeywords.java|51|54|94.44%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 162 / 182 (89.01%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalPurgeJob.java|3|7|42.86%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoKeywords.java|51|54|94.44%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/ServerMainShowPartitionsTest.java [hunk 1]
```diff
@@ -44,28 +42,26 @@
...
 
+import static io.questdb.test.AbstractCairoTest.sink;
 import static io.questdb.test.AbstractGriffinTest.assertCursor;
 import static io.questdb.test.griffin.ShowPartitionsTest.replaceSizeToMatchOS;
 import static io.questdb.test.griffin.ShowPartitionsTest.testTableName;
 import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
 import static io.questdb.test.tools.TestUtils.insertFromSelectPopulateTableStmt;
 
 @RunWith(Parameterized.class)
+@Ignore
```
Comment (ideoma) [inline core/src/test/java/io/questdb/test/ServerMainShowPartitionsTest.java:64]: Do you want to keep it ignored?

**Review by ideoma — DISMISSED** [review]

File: core/src/test/java/io/questdb/test/ServerMainShowPartitionsTest.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/test/ServerMainShowPartitionsTest.java:64]: nope, they should pass now, good spot 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 162 / 182 (89.01%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalPurgeJob.java|3|7|42.86%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoKeywords.java|51|54|94.44%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 162 / 182 (89.01%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalPurgeJob.java|3|7|42.86%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|31|39|79.49%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoKeywords.java|51|54|94.44%|
|:large_blue_circle:|io/questdb/std/str/Path.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableConverter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/VacuumColumnVersions.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocketWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogAlertSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/json/JsonLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryFileStore.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Pull request merged**